src/deferred.js

```diff
@@ -0,0 +1,21 @@
+
+var oldDeferred = jQuery.Deferred;
+
+jQuery.Deferred = function() {
+	var deferred = oldDeferred.apply( this, arguments );
+
+	// Don't add this method if the current jQuery doesn't provide it
```

Comment:
I think I was confusing myself;  I was making the false assumption that `.pipe` was being removed in 3.0, but I realize that isn't the case.  What you said about older versions is def true though